pkg/appgw/ingress_rules.go

```diff
@@ -32,18 +32,29 @@ func (c *appGwConfigBuilder) getListenersFromIngress(ingress *v1beta1.Ingress, e
 
 func (c *appGwConfigBuilder) processIngressRule(rule *v1beta1.IngressRule, ingress *v1beta1.Ingress, env environment.EnvVariables) (map[Port]interface{}, map[listenerIdentifier]listenerAzConfig) {
 	frontendPorts := make(map[Port]interface{})
+
+	// certificate from ingress TLS spec
 	ingressHostnameSecretIDMap := c.newHostToSecretMap(ingress)
+
 	listeners := make(map[listenerIdentifier]listenerAzConfig)
 
 	// Private IP is used when either annotation use-private-ip or USE_PRIVATE_IP env variable is true.
 	usePrivateIPFromAnnotation, _ := annotations.UsePrivateIP(ingress)
 	usePrivateIPForIngress := usePrivateIPFromAnnotation || env.UsePrivateIP == "true"
 
+	appgwCertName, _ := annotations.GetAppGwSslCertificate(ingress)
```

Comment:
Please add a validation to check if the cert exists for ingress in https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/pkg/controller/prune.go

Reply:
> Please add a validation to check if the cert exists for ingress in https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/pkg/controller/prune.go

added a new function pruneNoSslCertificate to filter a non-match ingress